lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -81,17 +81,18 @@ static bool shouldCreateClassImpl(igraph::InstanceGraphNode *node) {
 /// to the targeted operation.
 struct PathInfo {
   PathInfo() = default;
-  PathInfo(Operation *op, FlatSymbolRefAttr symRef,
+  PathInfo(Location loc, bool canBeInstanceTarget, FlatSymbolRefAttr symRef,
            StringAttr altBasePathModule)
-      : op(op), symRef(symRef), altBasePathModule(altBasePathModule) {
-    assert(op && "op must not be null");
+      : loc(loc), canBeInstanceTarget(canBeInstanceTarget), symRef(symRef),
+        altBasePathModule(altBasePathModule) {
     assert(symRef && "symRef must not be null");
   }
 
-  operator bool() const { return op != nullptr; }
+  /// The Location of the hardware component targeted by this path.
+  std::optional<Location> loc = std::nullopt;
```

Comment:
Just double checking, since we're not using lookup anymore, do we need the PathInfo to be default constructible?

Reply:
Hmm maybe not, I'm not sure if anything else still requires that.

Reply:
Yeah, default constructible is still required, at least for how we're using it when we add items with `try_emplace`. I will go ahead and merge this for now, and we could potentially keep reworking this.